fw/cache.c

```diff
@@ -2431,7 +2446,11 @@ __cache_add_node(TDB *db, TfwHttpResp *resp, unsigned long key)
 	T_DBG3("%s: db=[%p] resp=[%p], req=[%p], key='%lu', data_len='%ld'\n",
 	       __func__, db, resp, resp->req, key, data_len);
 
-	/* TODO #522: revalidate existing entries before inserting a new one. */
+	/*
+	 * Remove each entry with current key, even fresh entries. We assume
+	 * that if we at current place, no valid entries are exists.
+	 */
+	tdb_entry_remove(db, key, &tfw_cache_rec_eq_req, resp->req, false);
```

Comment:
Isn't this exactly the invalidation (removal) of previously cached responses as discussed in https://github.com/tempesta-tech/tempesta-test/pull/602/files#r1628991786 ?

In particular, I'm wondering if we evict old entries on a test like
```sh
while :; do curl --http2 -ik -o /dev/null -D /dev/stdout -H 'Cache-Control: max-age=2' -H 'Host: tempesta-tech.com' https://tempesta-tech.com/blog/building-your-own-wordpress-staging-with-tempesta-fw/ ; sleep 3; done
```

To understand whether this (or any similar scenario, which we can imagine) scenario floods our cache could you please introduce a couple of counters for `/proc/tempesta/perfstat`:

- `Cache objects` - the number of cached objects, must be decreased on the removal
- `Cache bytes` - number of bytes at the moment stored in the cache